source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -156,7 +157,12 @@ Http::FilterHeadersStatus CompressorFilter::decodeHeaders(Http::RequestHeaderMap
   }
 
   const auto& request_config = config_->requestDirectionConfig();
-  if (!end_stream && request_config.compressionEnabled() &&
+  const bool is_not_upgrade =
+      !Http::Utility::isUpgrade(headers) ||
+      !Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.enable_compression_without_content_length_header");
```

Comment:
Seems odd to have the value of `is_not_upgrade` be dependent on the runtime flag? Can we rename the variable to make a bit more sense?

Reply:
idea was that we would drop runtime flag eventually and wont need to rename. if you feel strong about it - i will try to rework this part